### PR TITLE
BugFix2.2.3

### PR DESCRIPTION
**Pull request by ifarzana**

### CODE REVIEW

File: lib/Report/DistributionReport.php [hunk 1]
```diff
@@ -257,6 +257,7 @@ public function setReportScheduleFormData()
     /** @inheritdoc */
     public function generateSavedReportName($filterCriteria)
     {
+        $saveAs = '';
```
Comment (dasgarner) [inline lib/Report/DistributionReport.php]: I might be wrong, sorry if I am, but I think we can still get to the end of this method without setting `$saveAs` to be anything other than empty string?

File: lib/Report/SummaryReport.php [hunk 2]
```diff
@@ -244,6 +244,7 @@ public function setReportScheduleFormData()
     public function generateSavedReportName($filterCriteria)
     {
 
+        $saveAs = '';
```
Comment (dasgarner) [inline lib/Report/SummaryReport.php]: Same as other instance

File: lib/Storage/MongoDbTimeSeriesStore.php [hunk 3]
```diff
@@ -533,46 +557,27 @@ public function getStats($filterBy = [])
...
-                    ]
-                ],
-            ];
-            $totalCursor = $collection->aggregate($totalQuery);
+            // Total
+            $result->totalCount = $total;
 
         } catch (\MongoDB\Exception\RuntimeException $e) {
             $this->log->error($e->getMessage());
+            throw new GeneralException($e->getMessage());
```
Comment (dasgarner) [inline lib/Storage/MongoDbTimeSeriesStore.php]: Do we really want to pass e->getMessage out to the user? Will that mean anything to them at this point?

We could pass them a user friendly message instead?

```
 throw new GeneralException(__('Sorry we encountered an error getting Proof of Play data, please consult your administrator'));
```

File: lib/Report/DistributionReport.php (hunk 1 above)
Comment (ifarzana) [reply lib/Report/DistributionReport.php]: yes, we can get to end and no error is thrown.

Comment (ifarzana) [reply lib/Report/DistributionReport.php]: does that mean I can remove the line $saveAs = '';  ?

**Pull request merged**